core/src/marker/markerManager.h

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+#include "scene/styleContext.h"
+#include "scene/drawRule.h"
+#include "util/ease.h"
+#include "util/fastmap.h"
+#include "util/types.h"
+#include <memory>
+#include <vector>
+
+namespace Tangram {
+
+class MapProjection;
+class Marker;
+class Scene;
+class StyleBuilder;
+
+class MarkerManager {
+
+public:
+
+    // Set the Scene object whose styling information will be used to build markers.
+    void setScene(std::shared_ptr<Scene> scene);
+
+    // Create a new, empty marker and return its ID. An ID of 0 indicates an invalid marker.
+    MarkerID add();
+
+    // Try to remove the marker with the given ID; returns true if the marker was found and removed.
+    bool remove(MarkerID markerID);
+
+    // Set the styling string for a marker; returns true if the marker was found and updated.
+    bool setStyling(MarkerID markerID, const char* styling);
+
+    // Set whether a marker should be visible; returns true if the marker was found and updated.
+    bool setVisible(MarkerID markerID, bool visible);
+
+    // Set a marker to a point feature at the given position; returns true if the marker was found and updated.
+    bool setPoint(MarkerID markerID, LngLat lngLat);
+
+    // Set a marker to a point feature at the given position; if the marker was previously set to a point, this
+    // eases from the old position to the new one over the given duration with the given ease type; returns true if
+    // the marker was found and updated.
+    bool setPointEased(MarkerID markerID, LngLat lngLat, float duration, EaseType ease);
+
+    // Set a marker to a polyline feature at the given position; returns true if the marker was found and updated.
+    bool setPolyline(MarkerID markerID, LngLat* coordinates, int count);
+
+    // Set a marker to a polygon feature at the given position; returns true if the marker was found and updated.
+    bool setPolygon(MarkerID markerID, LngLat* coordinates, int* counts, int rings);
+
+    // Update the zoom level for all markers; markers are built for one zoom level at a time so when the current zoom
+    // changes, all marker meshes are rebuilt.
+    bool update(int zoom);
+
+    // Remove and destroy all markers.
+    void removeAll();
+
+    const std::vector<std::unique_ptr<Marker>>& markers() const;
+
+private:
+
+    Marker* getMarkerOrNull(MarkerID markerID);
```

Comment:
Glad you commented on this because it's actually intentional and I think it's a very good practice to write explicitly descriptive names like this! It makes the operation trivial to understand when reading it in-line.